app/cdap/components/DirectiveInput/Components/InputPanel/index.tsx

```diff
@@ -0,0 +1,222 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Box, Typography } from '@material-ui/core';
+import MyDataPrepApi from 'api/dataprep';
+import { defaultFuseOptions } from 'components/DirectiveInput/constants';
+import { IDirectiveUsage } from 'components/DirectiveInput/types';
+import ee from 'event-emitter';
+import Fuse from 'fuse.js';
+import reverse from 'lodash/reverse';
+import Mousetrap from 'mousetrap';
+import React, { useEffect, useState } from 'react';
+import globalEvents from 'services/global-events';
+import NamespaceStore from 'services/NamespaceStore';
+import styled from 'styled-components';
+import uuidV4 from 'uuid/v4';
+import { grey } from '@material-ui/core/colors';
+import { IHeaderNamesList } from 'components/GridTable/types';
+import { getFormattedSyntax, getLastWordOfSearchItem } from 'components/DirectiveInput/utils';
+import SearchListItem from 'components/DirectiveInput/Components/SearchListItem';
+
+interface IInputPanelProps {
+  setDirectivesList: React.Dispatch<React.SetStateAction<[]>>;
+  isDirectiveSet: boolean;
+  columnNamesList: IHeaderNamesList[];
+  onSearchItemClick: (value: string) => void;
+  getDirectiveSyntax: (results: IDirectiveUsage[], value: boolean) => void;
+  inputDirective: string;
+}
+
+const SearchWrapper = styled(Box)`
+  display: block;
+`;
+
+const ResultRow = styled(Box)`
+  padding: 10px;
+  border-bottom: 1px solid ${grey[300]};
+  background-color: #ffffff;
+  &:hover {
+    background: #eff0f2;
+    cursor: pointer;
+  }
+`;
+
+const ActiveResultRow = styled(ResultRow)`
+  background-color: #eff0f2;
+`;
+
+export default function({
+  setDirectivesList,
+  isDirectiveSet,
+  columnNamesList,
+  onSearchItemClick,
+  getDirectiveSyntax,
+  inputDirective,
+}: IInputPanelProps) {
+  const [searchResults, setSearchResults] = useState<IDirectiveUsage[]>([]);
+  const [inputText, setInputText] = useState<string>('');
+  const [selectedIndex, setSelectedIndex] = useState<number | null>(null);
+  const eventEmitter = ee(ee);
+  const [fuse, setFuse] = useState(new Fuse([], { ...defaultFuseOptions }));
+
+  const getUsage = () => {
+    if (!isDirectiveSet) {
+      // If Directive name is not yet entered in input then the search list will be directive list else will be column list
+      MyDataPrepApi.getUsage({ context: NamespaceStore.getState().selectedNamespace }).subscribe(
+        (res) => {
+          setDirectivesList(res.values);
+          setFuse(new Fuse(res.values, { ...defaultFuseOptions }));
```

Comment:
Rather than keeping `Fuse` in state and creating a new one each time, it looks like you could create a `Fuse` instance and then update it with `setCollection` when the directives are changed. Did you look into that?

Reply:
Yes we tried using setCollection, but the fuse was not updating and no search list is shown, we need to keep it in setState to update fuse values

Reply:
Did you try `setCollection` with the spread operator? like `fuse.setCollection([...res.values])`